midend/local_copyprop.cpp

```diff
@@ -204,8 +210,34 @@ IR::AssignmentStatement *DoLocalCopyPropagation::preorder(IR::AssignmentStatemen
     return postorder(as);
 }
 
+/* Function to check if left and right side of
+ * assignment statement are equivalent.
+ * FIXME -- This function is temporary, proper deep comparison may be required */
+bool DoLocalCopyPropagation::equiv(const IR::Expression *left, const IR::Expression *right) {
+    // Compare names of variables (at this pass, all names are unique)
+    auto l1 = left->to<IR::PathExpression>();
+    auto r1 = right->to<IR::PathExpression>();
+    if (l1 && r1) {
+        return l1->path->name == r1->path->name;
+    }
+    // Compare references to arrays with indices
+    auto l2 = left->to<IR::Operation_Binary>();
+    auto r2 = right->to<IR::Operation_Binary>();
+    if (l2 && r2) {
+        return equiv(l2->left, r2->left) && equiv(l2->right, r2->right);
+    }
+    // Compare packet header/metadata fields
+    auto l3 = left->to<IR::Member>();
+    auto r3 = right->to<IR::Member>();
+    if (l3 && r3) {
+        return l3->member == r3->member && equiv(l3->expr, r3->expr);
+    } else {
+        return false;
```

Comment:
Could any other `Operation_Unary` other than `IR::Member` appear on the LHS of an assignment?